### PR TITLE
Sign in

### PR DESCRIPTION
**Pull request by tolia-ogurt**

### CODE REVIEW

File: app/src/main/java/com/example/trempel/SessionManager.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.example.trempel
+
+import android.content.Context
+import android.content.SharedPreferences
+import javax.inject.Inject
+
+class SessionManager @Inject constructor(context: Context) {
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/SessionManager.kt]: `internal`?

File: app/src/main/java/com/example/trempel/SessionManager.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        Context.MODE_PRIVATE
+    )
+
+    fun saveAuth(token: String) {
+        val editor = prefs.edit()
+        editor.putString(USER_TOKEN, token)
+        editor.apply()
+    }
+
+    fun fetchAuthToken(): String? {
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/SessionManager.kt]: can you add the variable `authToken` and move `saveAuth` and `fetchAuthToken` methods to set and get accessors?

File: app/src/main/java/com/example/trempel/SessionManager.kt [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        return prefs.getString(USER_TOKEN, null)
+    }
+
+    companion object {
+        const val USER_TOKEN = "user_token"
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/SessionManager.kt:22]: `private`?

File: app/src/main/java/com/example/trempel/SignInRepository.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import com.example.trempel.network.model.LoginRequest
+import com.example.trempel.network.model.LoginResponse
+import com.example.trempel.network.service.SignInService
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.schedulers.Schedulers
+import javax.inject.Inject
+
+class SignInRepository @Inject constructor(
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/SignInRepository.kt]: `internal`?

File: app/src/main/java/com/example/trempel/network/model/LoginRequest.kt [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.example.trempel.network.model
+
+data class LoginRequest(
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/network/model/LoginRequest.kt]: `internal`?

File: app/src/main/java/com/example/trempel/network/model/LoginResponse.kt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package com.example.trempel.network.model
+
+data class LoginResponse(
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/network/model/LoginResponse.kt]: `internal`?

File: app/src/main/java/com/example/trempel/network/service/SignInService.kt [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package com.example.trempel.network.service
+
+import com.example.trempel.network.model.LoginRequest
+import com.example.trempel.network.model.LoginResponse
+import io.reactivex.Single
+import retrofit2.http.Body
+import retrofit2.http.FormUrlEncoded
+import retrofit2.http.POST
+
+interface SignInService {
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/network/service/SignInService.kt]: `internal`?

File: app/src/main/java/com/example/trempel/ui/main_activity/MainViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,13 @@
+package com.example.trempel.ui.main_activity
+
+import androidx.lifecycle.ViewModel
+import com.example.trempel.SessionManager
+import javax.inject.Inject
+
+class MainViewModel @Inject constructor(
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/ui/main_activity/MainViewModel.kt]: `internal`?

File: app/src/main/res/layout/sign_in_fragment.xml [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+            name="viewModel"
+            type="com.example.trempel.ui.login.SignInViewModel" />
+    </data>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:background="@color/black">
+
+        <View
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/sign_in_fragment.xml]: Why not ImageView?

File: app/src/main/res/layout/sign_in_fragment.xml [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+            android:id="@+id/trempel_logo"
+            android:layout_width="210dp"
+            android:layout_height="210dp"
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/sign_in_fragment.xml]: dimensions should be extracted

File: app/src/main/res/layout/sign_in_fragment.xml [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+            android:fontFamily="@font/montserrat_light"
+            android:onClick="@{()-> viewModel.login()}"
+            android:text="@string/btn_login"
+            android:textSize="@dimen/normal_text_size"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="@id/til_password"
+            app:layout_constraintStart_toStartOf="@id/til_password"
+            app:layout_constraintTop_toBottomOf="@id/til_password" />
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (DaryaErofeeva) [inline app/src/main/res/layout/sign_in_fragment.xml:124]: Not sure that this layout fits to smaller/bigger screen size, it'd be better to use constraint weight/chains/barriers for it

File: app/src/main/java/com/example/trempel/SessionManager.kt (hunk 1 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/SessionManager.kt]: done

File: app/src/main/java/com/example/trempel/SignInRepository.kt (hunk 4 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/SignInRepository.kt]: done

File: app/src/main/java/com/example/trempel/SessionManager.kt (hunk 3 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/SessionManager.kt:22]: done

File: app/src/main/java/com/example/trempel/SessionManager.kt (hunk 2 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/SessionManager.kt]: done

File: app/src/main/java/com/example/trempel/network/model/LoginRequest.kt (hunk 5 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/network/model/LoginRequest.kt]: done

File: app/src/main/java/com/example/trempel/network/model/LoginResponse.kt (hunk 6 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/network/model/LoginResponse.kt]: done

File: app/src/main/java/com/example/trempel/network/service/SignInService.kt (hunk 7 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/network/service/SignInService.kt]: done

File: app/src/main/java/com/example/trempel/ui/main_activity/MainViewModel.kt (hunk 8 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/ui/main_activity/MainViewModel.kt]: done

File: app/src/main/res/layout/sign_in_fragment.xml (hunk 9 above)
Comment (tolia-ogurt) [reply app/src/main/res/layout/sign_in_fragment.xml]: done

File: app/src/main/res/layout/sign_in_fragment.xml (hunk 10 above)
Comment (tolia-ogurt) [reply app/src/main/res/layout/sign_in_fragment.xml]: done

File: app/src/main/java/com/example/trempel/ui/categories/CategoryFragment.kt [hunk 12]
```diff
@@ -41,4 +41,6 @@ internal class CategoryFragment : Fragment(R.layout.category_fragment) {
             Toast.makeText(this.context, it, Toast.LENGTH_SHORT).show()
         })
     }
+
+
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/ui/categories/CategoryFragment.kt]: redundant change

File: app/src/main/java/com/example/trempel/di/NetworkModule.kt [hunk 13]
```diff
@@ -35,4 +36,9 @@ internal class NetworkModule {
     fun provideCategoryService(retrofit: Retrofit): CategoryService {
         return retrofit.create(CategoryService::class.java)
     }
+    @Provides
```
Comment (DaryaErofeeva) [inline app/src/main/java/com/example/trempel/di/NetworkModule.kt:40]: I'd prefer to have an empty line between functions

File: app/src/main/java/com/example/trempel/ui/categories/CategoryFragment.kt (hunk 12 above)
Comment (tolia-ogurt) [reply app/src/main/java/com/example/trempel/ui/categories/CategoryFragment.kt]: done

**Review by DaryaErofeeva — APPROVED** [review]

**Pull request merged**